modules/component-store/src/map-response.ts

```diff
@@ -0,0 +1,39 @@
+import { EMPTY, Observable } from 'rxjs';
+
+import { catchError, tap } from 'rxjs/operators';
+
+/**
+ * Maps the response in ComponentStore effects in a safe way, without
+ * additional boilerplate.
+ * It enforces that the error case is handled and that the effect would still be
+ * running should an error occur.
+ *
+ * Takes an optional third argument for a `complete` callback.
+ *
+ * ```typescript
+ * readonly dismissedAlerts = this.effect<Alert>(alert$ => {
+ *  return alert$.pipe(
+ *      concatMap(
+ *          (alert) => this.alertsService.dismissAlert(alert).pipe(
+ *              mapResponse(
+ *                 (dismissedAlert) => this.alertDismissed(dismissedAlert),
+ *                 (error) => this.logError(error),
+ *              ))));
+ *   });
+ * ```
+ */
+export function mapResponse<T>(
+  nextFn: (next: T) => void,
+  errorFn: (error: unknown) => void,
+  completeFn?: () => void
+): (source: Observable<T>) => Observable<T> {
```

Comment:
I suggest to add two signatures for this function, similar to `tap` operator in order to make it more flexible.

```typescript
export function tap<T>(next?: (x: T) => void, error?: (e: any) => void, complete?: () => void): MonoTypeOperatorFunction<T>;
export function tap<T>(observer: PartialObserver<T>): MonoTypeOperatorFunction<T>;
```

Reply:
The goal is to not make it flexible. I want to force to handle the error.

Reply:
Okay, I suggested that based on the implementation. According to the operator implementation too, I suggest the name **safeTap** or something similar. 🙂

Reply:
Thanks @markostanimirovic that's a good suggestion.
We had similar discussions about `safeCall` operator for `@ngrx/effect` and eventually called it `mapToAction` and then renamed to `act`: https://github.com/ngrx/platform/issues/1224

Looking at the poll `tapResponse` appears to be the clear winner.

@beeman, technically this operator still `maps`, but only the error case :)

Reply:
@alex-okrushko 
Naming - the hardest thing 😄

I would also choose `tapResponse` between `tapResponse`, `mapResponse` and `handleResponse`.

`Response` as a part of name confused me a little bit, because it associates me with HTTP response 🙂